### DISCUSSION TITLE
Integrating Headless UI with an existing vanilla, frameworkless PHP site

### DISCUSSION THREAD

**Discussion by waj00115** [tailwindlabs/tailwindcss#8413] [Help]
Hi everyone,

Looking for advise and this may very well be more of a node question than anything else.

I have a vanilla PHP site that's fairly hefty in size. It was largely developed back in 2006, and has no framework in use. Other than some tweaks here and there for changes to the PHP language, probably 98% of the PHP code is in tact. It's not a viable project to grab to convert to an entirely Vue/React site. The PHP aside, I did convert it over several years ago to use Tailwind and handle the JS with jQuery.

I would like to integrate Headless UI in to better handle the interaction, but truly redoing it as a Vue/React site just isn't in the cards right now. Anybody have any tips, tutorials, etc. to best accomplish this? I think I'm getting caught up in the build process more than anything else right now, and how to integrate in a fully-PHP site in with that.

Thanks!

**Comment by miquelbrazil** [comment]
Sorry for the necropost. I ran across your discussion while researching approaches for building a sort of "componentization" architecture into my PHP-based SaaS codebase. For about a year now, I've been simmering with inspiration from CakePHP's Widgets architecture, Blade with Livewire's file-based markup structure and Plates API for inserting these components into markup.

The road that led me here is that I've been a huge fan of @calebporzio's [AlpineJS](https://github.com/alpinejs/alpine). AlpineJS is the underlying JS library that [Livewire](https://github.com/livewire/livewire) is built on top of and used in Laravel Blade. Livewire is dope at creating "reactivity" between page elements (or components) without having to also concern yourself with the presentation logic, like with React and friends. But Livewire is heavily tied to how Blade works, and I don't use Laravel (my project uses [Slim 4](https://github.com/slimphp/Slim) for routing and build the rest from scratch or on top of Symfony components). AlpineJS fits the bill to bring those Livewire features that let you do some pretty amazing interactive stuff by just rendering out JSON data with an element you want to behave reactively in the DOM. Have you tried it on your project? It also plays nicely with Tailwind.
